File: server/crypto.py
from Crypto.Cipher import AES
from Crypto.Util.Padding import unpad
import base64
import hashlib

def get_key(user_id, secret):
    raw = f'eliano-key-{user_id}-{secret}'
    return hashlib.sha256(raw.encode()).digest()  # 32 bytes

def decrypt_email(encrypted_b64, user_id, secret):
    data = base64.b64decode(encrypted_b64)
    iv = data[:16]
    ciphertext = data[16:]

    key = get_key(user_id, secret)
    cipher = AES.new(key, AES.MODE_CBC, iv)
    decrypted = cipher.decrypt(ciphertext)
    return unpad(decrypted, AES.block_size).decode('utf-8')

# Teste
user_id = 1
secret = 'default-secret'  # deve ser igual ao process.env.ENCRYPTION_SECRET ou o valor padrão

# Cole aqui o valor gerado no TS
encrypted_data = 'oaMSZG2Z7nxSMkUpBQVr9oOrVu3+5Y5aO/g3rdECUp4='  # exemplo: '9I5Mu3Ivg8HRKXLvU4TfCtv1CZ05xslrxKncz+NFFhA='
print(decrypt_email(encrypted_data, user_id, secret))

File: server/crypto.ts
import CryptoJS from 'crypto-js';

// Função que retorna uma chave de 32 bytes (256 bits) em formato Hex
const getEncryptionKey = (userId: number): CryptoJS.lib.WordArray => {
  const rawKey = `eliano-key-${userId}-${process.env.ENCRYPTION_SECRET || 'default-secret'}`;
  const hash = CryptoJS.SHA256(rawKey); // Garante 256 bits
  return hash; // WordArray de 32 bytes
};

export const encryptEmail = (data: string, userId: number): string => {
  const key = getEncryptionKey(userId);
  const iv = CryptoJS.lib.WordArray.random(16); // 16 bytes = 128 bits

  const encrypted = CryptoJS.AES.encrypt(data, key, {
    iv,
    mode: CryptoJS.mode.CBC,
    padding: CryptoJS.pad.Pkcs7,
  });

  const result = iv.concat(encrypted.ciphertext).toString(CryptoJS.enc.Base64);
  return result;
};

export const decryptEmail = (encryptedData: string, userId: number): string => {
  try {
    if (!encryptedData || encryptedData.length < 24 || !/^[A-Za-z0-9+/=]+$/.test(encryptedData)) {
      return encryptedData;
    }
    const raw = CryptoJS.enc.Base64.parse(encryptedData);
    const iv = CryptoJS.lib.WordArray.create(raw.words.slice(0, 4), 16); // 4 words = 16 bytes
    const ciphertext = CryptoJS.lib.WordArray.create(raw.words.slice(4), raw.sigBytes - 16);

    const key = getEncryptionKey(userId);
    const cipherParams = CryptoJS.lib.CipherParams.create({ ciphertext });
    const decrypted = CryptoJS.AES.decrypt(cipherParams, key, {
      iv,
      mode: CryptoJS.mode.CBC,
      padding: CryptoJS.pad.Pkcs7,
    });

    return decrypted.toString(CryptoJS.enc.Utf8);
  } catch (err) {
    console.error('Erro ao descriptografar:', err);
    return encryptedData;
  }
};